model/src/main/kotlin/ScannerRun.kt

```diff
@@ -202,3 +265,36 @@ private fun scanResultForProvenanceResolutionIssue(packageProvenance: KnownProve
 
 private fun ScanSummary.addIssue(issue: Issue?): ScanSummary =
     if (issue == null) this else copy(issues = (issues + issue).distinct())
+
+private fun mergeFileLists(fileListByPath: Map<String, FileList>): FileList {
+    require(fileListByPath.containsKey("")) {
+        "There must be a file list associated with the empty path."
+    }
+
+    val provenance = fileListByPath.getValue("").provenance
```

Comment:
So this could actually be combined with the above requirement:

    val provenance = requireNotNull(fileListByPath[""]?.provenance) { ... }